src/main/java/org/opentripplanner/routing/algorithm/mapping/AlertToLegMapper.java

```diff
@@ -61,43 +57,54 @@ public void addTransitAlertsToLeg(Leg leg, boolean isFirstLeg) {
     LocalDate serviceDate = leg.getServiceDate();
 
     if (fromStop instanceof RegularStop stop) {
-      Collection<TransitAlert> alerts = getAlertsForStopAndRoute(stop, routeId);
-      alerts.addAll(getAlertsForStopAndTrip(stop, tripId, serviceDate));
-      alerts.addAll(getAlertsForRelatedStops(stop, transitAlertService::getStopAlerts));
-      addTransitAlertsToLeg(leg, departingStopConditions, alerts, legStartTime, legEndTime);
+      Set<StopCondition> stopConditions = isFirstLeg
+        ? StopCondition.DEPARTURE
+        : StopCondition.FIRST_DEPARTURE;
```

Comment:
Isn't this logic upside down? I would expect first leg to use `StopCondition.FIRST_DEPARTURE`.